### DISCUSSION TITLE
Is there a way to produce static builds of the tools?

### DISCUSSION THREAD

**Discussion by SamusAranX** [mamedev/discussions#138] [Q&A]
Specifically I need a way to produce builds that will run on a dumbed-down x64 linux device that I can't install the dependencies on.

Is this something that's officially supported? I couldn't find anything of value online.

**Comment by galibert** [comment]
I'd try make ARCHOPTS="-static".

**Reply by SamusAranX** [reply]
Mega late reply, but I just got to try this.
I just get an error about the linker not finding SDL2:
```
$ make ARCHOPTS="-static" TOOLS=1 EMULATOR=0 NO_OPENGL=1 NO_USE_PORTAUDIO=1 NO_USE_PULSEAUDIO=1
GCC 14.2.1 detected
Linking castool...
/usr/bin/ld: cannot find -lSDL2: No such file or directory
collect2: error: ld returned 1 exit status
make[2]: *** [castool.make:254: ../../../../../castool] Error 1
make[1]: *** [Makefile:139: castool] Error 2
make: *** [makefile:1290: linux_x64] Error 2
```

I'm on Fedora 40 and the `SDL2-devel` package is installed. Not sure what else it could need.

**Reply by cuavas** [reply]
Do you have a statically linked static library for SDL2 installed?  If you want a static build you need static versions of all its dependencies, too.

**Reply by SamusAranX** [reply]
Where would I get those?

**Reply by cuavas** [reply]
Try installing SDL2-static? https://packages.fedoraproject.org/pkgs/SDL2/SDL2-static/index.html

If it still doesn’t work, you probably need to compile it yourself.  What you’re doing is pretty niche.

**Reply by SamusAranX** [reply]
I installed `SDL2-static`, `libstdc++-static`, and `glibc-static` and that got me past the initial errors, but now I'm stuck on this with no clear remedy:

```
Linking chdman...
/usr/bin/ld: ../../../../linux_gcc/bin/x64/Release/mame_mame/libocore_sdl.a(osdlib_unix.o): in function `osd::(anonymous namespace)::dynamic_module_posix_impl::get_symbol_address(char const*)':
osdlib_unix.cpp:(.text+0x4a): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: ../../../../linux_gcc/bin/x64/Release/mame_mame/libocore_sdl.a(posixsocket.o): in function `posix_open_socket(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, std::unique_ptr<osd_file, std::default_delete<osd_file> >&, unsigned long&)':
posixsocket.cpp:(.text+0x2bb): warning: Using 'gethostbyname' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
Compiling src/devices/cpu/8x300/8x300dasm.cpp...
cc1plus: error: one or more PCH files were found, but they were invalid
cc1plus: note: use -Winvalid-pch for more information
<command-line>: fatal error: ../../../../linux_gcc/obj/x64/Release/emu.h: No such file or directory
compilation terminated.
make[2]: *** [dasm.make:1934: ../../../../linux_gcc/obj/x64/Release/src/devices/cpu/8x300/8x300dasm.o] Error 1
make[1]: *** [Makefile:100: dasm] Error 2
make: *** [makefile:1290: linux_x64] Error 2
```

Is this something that's even fixable?

**Reply by cuavas** [reply]
That suggests you updated a runtime library or changed important compile options without doing a clean build.  Blow away the build folder and try again.

**Reply by SamusAranX** [reply]
Did that, reran make, and it produced a bunch of static builds no problem.
Thanks for all your help!
